### PR TITLE
Prep for wayland

### PR DESCRIPTION
**Pull request by wmww**
Update a bunch of files in preparation for Wayland support. As with my other PRs you can't test Wayland support yet, just make sure it doesn't break anything on X11.

### CODE REVIEW

**Comment by wmww** [comment]
Rebased on top of latest master

File: mate-panel/panel-struts.c [hunk 1]
```diff
@@ -259,6 +264,8 @@ panel_struts_set_window_hint (PanelToplevel *toplevel)
 
 	widget = GTK_WIDGET (toplevel);
 
+	g_return_if_fail(GDK_IS_X11_DISPLAY (gtk_widget_get_display (widget)));
```
Comment (raveit65) [inline mate-panel/panel-struts.c]: I think you love not to use a space before a bracket :)
Can you fixed that here and in other occurrences in this PR, please.

I know that we have such formatting issues in existing code........

Comment (wmww) [reply mate-panel/panel-struts.c]: I'm sorry that MATE and GTK code insists on a space where it doesn't belong. I'll fix 'em

Comment (wmww) [reply mate-panel/panel-struts.c]: Done

**Pull request merged**